pkg/webhook/resources/setting/validator.go

```diff
@@ -276,6 +283,51 @@ func validateSSLCertificates(setting *v1beta1.Setting) error {
 	return nil
 }
 
+func validateSSLParameters(setting *v1beta1.Setting) error {
+	if setting.Value == "" {
+		return nil
+	}
+
+	sslParameter := &settings.SSLParameter{}
+	if err := json.Unmarshal([]byte(setting.Value), sslParameter); err != nil {
+		return werror.NewInvalidError(err.Error(), "value")
+	}
+
+	if sslParameter.Protocols == "" && sslParameter.Ciphers == "" {
+		return nil
+	}
+
+	if err := validateSSLProtocols(sslParameter); err != nil {
+		return werror.NewInvalidError(err.Error(), "protocols")
+	}
+
+	// TODO: Validate ciphers
+	// Currently, there's no easy way to actually tell what Ciphers are supported by rke2-ingress-nginx,
+	// we need to tell users where to look for ciphers in docs.
+	return nil
+}
+
+func validateSSLProtocols(param *settings.SSLParameter) error {
+	if param.Protocols == "" {
+		return nil
+	}
+
+	for _, given := range strings.Split(param.Protocols, " ") {
+		supported := false
+		for _, supportedProtocol := range supportedSSLProtocols {
+			if given == supportedProtocol {
+				supported = true
+				break
+			}
+		}
```

Comment:
Can be simplified using `slice.ContainsString(supportedSSLProtocols, given)`

Reply:
Great tip!